### PR TITLE
Support for Event Grid Domain/Topic system managed identity

### PR DESCRIPTION
**Pull request by jrauschenbusch**
Fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/9525

```sh
make acctests SERVICE='eventgrid' TESTARGS='-run=TestAccEventGridTopic_basicWithIdentity' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/eventgrid/tests/ -run=TestAccEventGridTopic_basicWithIdentity -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccEventGridTopic_basicWithIdentity
=== PAUSE TestAccEventGridTopic_basicWithIdentity
=== CONT  TestAccEventGridTopic_basicWithIdentity
--- PASS: TestAccEventGridTopic_basicWithIdentity (116.60s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/eventgrid/tests     118.483s
```

```sh
make acctests SERVICE='eventgrid' TESTARGS='-run=TestAccEventGridDomain_basicWithIdentity' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/eventgrid/tests/ -run=TestAccEventGridDomain_basicWithIdentity -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccEventGridDomain_basicWithIdentity
=== PAUSE TestAccEventGridDomain_basicWithIdentity
=== CONT  TestAccEventGridDomain_basicWithIdentity
--- PASS: TestAccEventGridDomain_basicWithIdentity (119.35s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/eventgrid/tests     121.912s
```

### CODE REVIEW

File: website/docs/r/eventgrid_topic.html.markdown [hunk 1]
```diff
@@ -56,6 +58,14 @@ The following arguments are supported:
 
 ---
 
+A `identity` block exports the following:
```
Comment (mbfrahry) [inline website/docs/r/eventgrid_topic.html.markdown:61]:
```suggestion
A `identity` block supports the following:
```

We should also call out `identity_ids`

And in the export section of the docs, we want to call out `principal_id` and `tenant_id`

File: website/docs/r/eventgrid_domain.html.markdown [hunk 2]
```diff
@@ -54,6 +56,14 @@ The following arguments are supported:
 
 ---
 
+A `identity` block exports the following:
```
Comment (mbfrahry) [inline website/docs/r/eventgrid_domain.html.markdown:59]: The same doc note should be applied here

File: azurerm/internal/services/eventgrid/eventgrid.go [hunk 3]
```diff
@@ -93,3 +133,65 @@ func flattenInboundIPRules(in *[]eventgrid.InboundIPRule) []interface{} {
...
+
+	identityIdsRaw := raw["identity_ids"].(*schema.Set).List()
+	identityIds := make(map[string]*eventgrid.UserIdentityProperties)
+	for _, v := range identityIdsRaw {
+		identityIds[v.(string)] = &eventgrid.UserIdentityProperties{}
+	}
+
+	if len(identityIds) > 0 {
+		if identity.Type != eventgrid.IdentityTypeUserAssigned && identity.Type != eventgrid.IdentityTypeSystemAssignedUserAssigned {
+			return nil, fmt.Errorf("`identity_ids` can only be specified when `type` includes `UserAssigned`")
```
Comment (mbfrahry) [inline azurerm/internal/services/eventgrid/eventgrid.go:158]: We should have a test where we use `identity_ids` as well

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @jrauschenbusch! This is looking mostly good but we need to flesh out the docs a little and add another test

**Comment by katbyte** [comment]
Hey @jrauschenbusch - as theres been no activity on this since may i'm going to close it. Please do reopen it when you have time to address the comments! thanks

**Pull request closed without merging**